### ISSUE TITLE
Fails to bundle Fonts in .scss file

### ISSUE THREAD

**Issue by wsfuller** [webpack-contrib/sass-loader#411]
**Directory Setup**

    /src
        index.jsx
        index.scss
        /assets
            /fonts
                font.eot
                ...

---

**index.jsx**

require('./index.scss');
...

---

**index.scss**

    @font-face {
      font-family: 'Air-Bold';
      src: url('/assets/fonts/Air-Bold.eot');
      src: url('/assets/fonts/Air-Bold.eot?#iefix') format('embedded-opentype'),
           url('/assets/fonts/Air-Bold.woff') format('woff'),
           url('/assets/fonts/Air-Bold.ttf')  format('truetype'),
           url('/assets/fonts/Air-Bold.svg') format('svg');
      font-style: normal;
      font-weight:  700;
    }
 
    body{ font-family: 'Air-Bold', serif; }

---

**webpack.config**

    module.exports = {
      entry: './src/index.jsx',
      output: {
        path: path.resolve(__dirname, "dist"),
        filename: './bundle.js',
        publicPath: '/dist/assets/'
      },
      module: {
        rules: [
          {
            test: /\.(s?)css$/,
            use: ExtractTextPlugin.extract({
                fallback: 'style-loader',
                use: ['css-loader', 'sass-loader', 'resolve-url-loader', 'postcss-loader'],
                publicPath: '/dist'
            })
          },
          {
            test: /\.(js|jsx)$/,
            exclude: /node_modules/,
            use: "babel-loader"
          },
          {
            test: /\.(eot|svg|ttf|woff|woff2)$/,
            exclude: /node_modules/,
            loader: 'file-loader'
          }
        ]
      },
      devServer:{
        contentBase: path.join(__dirname, "dist");
      },
      plugins: [
        new webpack.HotModuleReplacementPlugin(),
        new HtmlWebpackPlugin({
          template: './src/index.ejs'
        }),
        new ExtractTextPlugin({
          filename: 'app.css',
          disable: false,
          allChunks: true
        }),
        new FaviconsWebpackPlugin({
            ...
            }
        })
      ]
    };

---

**Dependencies**

* "css-loader": "~0.26.2",
* "file-loader": "^0.10.1",
* "node-sass": "~4.5.0",
* "postcss-loader": "~1.3.3",
* "postcss-smart-import": "~0.6.9",
* "precss": "~1.4.0",
* "sass-loader": "~6.0.2",
* "style-loader": "~0.13.2",
* "url-loader": "^0.5.8",
* "webpack": "^2.2.1",
* "webpack-dev-server": "~2.4.1"

The problem I'm getting is simply 404s on the fonts. I've added the `resolve-url-loader`, have gone to several of resources including:

* [Fonts with webpack](https://shellmonger.com/2016/01/22/working-with-fonts-with-webpack/)
* [SO: How to copy static files to build directory with Webpack](http://stackoverflow.com/questions/27639005/how-to-copy-static-files-to-build-directory-with-webpack)
* [Youtube Building fonts](https://www.youtube.com/watch?v=5GV7HDDf9Yg)

Nothing seems to be working. 

There's also nothing documenting how to achieve this, that I can see reading and re-reading, in the [Official Webpack 2 Docs](https://webpack.js.org/loaders/sass-loader/#extracting-style-sheets).

Running `webpack` everything compiles correct but fonts are never carried over to `/dist/assets` running the `webpack-dev-server` the font files are never found.

**Comment by xXanth0s** [comment]
Have the same problem :(

**Comment by evilebottnawi** [comment]
@xXanth0s can you create minimal reproduce test repo? it is allow to resolve issue more faster :smile:

**Comment by wsfuller** [comment]
@xXanth0s 

To fix this I did the following:

1. Created Font Loader Variable (**before** `module.exports={...}`)
    * `var fontLocal = 'file-loader?name=[name].[ext]`
    * `var fontBundle = 'file-loader?name=assets/fonts/[name].[ext]`
2. Created ternary operator (**before** `module.exports={...}`)
    * `var fontConfig = isBundle ? fontBundle : fontLocal;`
3. Update loader to use `fontConfig` (**inside** `module.exports={...}`)
    * `test: /\.(eot|ttf|woff|woff2)$/, loader: fontConfig`


For clarification, I use **"Local"** for **"Development" or "Dev"** and **"Bundle"**  for **"Production" or "Prod" or "Build"**. There're specific reasons for this, just so that's clear. 

The other possibility (and this was another issue I had) was loader ordering. So in the documentation I never found where it said the order in which to place your loaders. Common sense would be left to right for first to last in a Western English speaking world. In UNIX it's the opposite for whatever reason there is for that (would love to know more about this). There's a SO article that I was trying to find discussing this but unable to turn it up, where I was able to get some high level answers. 

So my new, working, ordering for my CSS loaders is `use: ['css-loader','resolve-url-loader','postcss-loader','sass-loader']`


Hopefully this answers the issue you were having as well or at least helps.
